### PR TITLE
fix: tooltip issue

### PR DESCRIPTION
**Pull request by niloofar-deriv**

### CODE REVIEW

File: src/components/Tooltip/index.tsx [hunk 1]
```diff
@@ -1,29 +1,55 @@
...
+const TooltipArrowVariantClass = {
+    dark: "deriv-arrow--dark",
+    error: "deriv-arrow--error",
+    general: "deriv-arrow--general",
+} as const;
+
 export const Tooltip = ({
     children,
     className,
+    activeClassName,
```
Comment (shayan-deriv) [inline src/components/Tooltip/index.tsx:45]: what is this  for? 🤔

File: src/components/Tooltip/index.tsx [hunk 2]
```diff
@@ -1,29 +1,55 @@
...
-type TooltipProps = {
-    children?: ReactNode;
-    className?: string;
-    message: ReactNode;
-    position?: TooltipPositionType;
-    triggerAction?: TooltipTriggerActionType;
-    variant?: TooltipVariantType;
+type TTooltip = Omit<
+    ComponentProps<"div">,
+    "ref" | "onMouseEnter" | "onMouseLeave"
```
Comment (shayan-deriv) [inline src/components/Tooltip/index.tsx:14]: why removing these? 🤔

Comment (shayan-deriv) [reply src/components/Tooltip/index.tsx:14]: we should pass all possible props if we are extending the types from div element

Comment (niloofar-deriv) [reply src/components/Tooltip/index.tsx:14]: Cause we are handling them internally and user dose not need to change these

File: src/components/Tooltip/index.tsx (hunk 1 above)
Comment (niloofar-deriv) [reply src/components/Tooltip/index.tsx:45]: For the message wrapper maybe they need to add sth to it

File: src/components/Tooltip/index.tsx [hunk 3]
```diff
@@ -1,29 +1,55 @@
...
     className,
+    activeClassName,
     message,
     position = "top",
     triggerAction = "hover",
     variant = "general",
-}: TooltipProps) => {
+    onClick,
+    ...rest
+}: PropsWithChildren<TTooltip>) => {
```
Comment (aizad-deriv) [inline src/components/Tooltip/index.tsx:52]: let's add a forwardref for this component aswell 😃

**Pull request closed without merging**